src/main/scala/carldata/series/Session.scala

```diff
@@ -2,3 +2,29 @@ package carldata.series
 
 case class Session(startIndex: Int, endIndex: Int) {
 }
+
+object Session {
+  def findSessions[V: Fractional](ts: TimeSeries[V]): Seq[Session] = {
+    val tss = ts.values.zipWithIndex //time series values indexed
+    val zps: Seq[((V, Int), (V, Int))] = tss.zip(tss.tail).dropWhile(x => x._1._1 == 0) //zipped pairs
```

Comment:
dropWhile is unnecessary here. It doesn't improve algorithm, but forces someone to understand why it is used here.